executor/executor.go

```diff
@@ -67,23 +66,27 @@ func New(sdAPI api.API, args []string) (Executor, error) {
 
 func execCommand(path string, args []string) (err error) {
 	cmd := command(path, args...)
+	errChan := make(chan error, 1)
 	if !terminal.IsTerminal(syscall.Stdin) {
 		stdin, err := cmd.StdinPipe()
 		if err != nil {
 			lgr.Debug.Printf("failed to open StdinPipe: %v", err)
 			return err
 		}
-		b, err := ioutil.ReadAll(os.Stdin)
-		if err != nil {
-			lgr.Debug.Printf("failed to read Stdin: %v", err)
-			return err
-		}
-		_, err = io.WriteString(stdin, string(b))
-		if err != nil {
-			lgr.Debug.Printf("failed to write StdinPipe: %v", err)
-			return err
-		}
-		stdin.Close()
+		go func() {
+			defer stdin.Close()
+			defer close(errChan)
```

Comment:
I guess `errChan` is also closed soon if this goroutine ends soon with a small stdin. Then, closed `<-errChan` may return `nil` even if there is an error.

Reply:
Never mind, It looks ok.
https://play.golang.org/p/rzsqkFMXqhV